File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    const result = [];
    privacies.forEach((privacy, i) => {
        const [dateStr, term] = privacy.split(' ');
        let date = new Date(dateStr);
        const month = terms.find((t) => t[0] === term).split(' ')[1]
        date.setMonth(date.getMonth() + Number(month));
        
        if (new Date(today) >= date) {
            result.push(i + 1);
        }
    })
    
    return result;
}

File: 프로그래머스/1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
import datetime
from dateutil.relativedelta import relativedelta

def solution(today, terms, privacies):
    answer = []
    
    # datetime.datetime.strptime(date_string, format)
    # : 문자열을 지정된 형식에 맞게 datetime 객체로 변환
    today = datetime.datetime.strptime(today, '%Y.%m.%d')
    
    for index, privacy in enumerate(privacies):
        start_date, term = privacy.split(' ')
        start_date = datetime.datetime.strptime(start_date, '%Y.%m.%d')
        
        month = findTerm(term, terms)
        
        end_date = start_date + relativedelta(months = month)
        
        if end_date <= today:
            answer.append(index + 1)
    
    return answer

def findTerm(term, terms):
    for item in terms:
        [_term, num] = item.split(' ')
        if _term == term:
            return int(num)

        
    
